### ISSUE TITLE
Remove linker-specific attributes in 'exe' mode

### ISSUE THREAD

**Issue by jtschuster** [dotnet/linker#3038]
The linker doesn't remove DynamicallyAccessedMembers, RequiresUnreferencedCode, etc. If we link in 'exe' mode, is there a reason we need to keep them? In library mode it makes sense, but I can't think of why we would need them for a published application.

**Comment by jtschuster** [comment]
@sbomer pointed out there is a LinkAttributes.xml ([link](https://github.com/dotnet/runtime/blob/c7eb44d11bb5acca6ccc40de42314663cbaff145/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml#L119-L122)) that removes these and can be enabled with `_AggressiveAttributeTrimming` property ([link](https://github.com/dotnet/sdk/blob/990a1a0809e3a843263bdb8d33e34d3801d44805/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L399-L402)), but it doesn't seem to be enabled anywhere in the sdk.

Perhaps this should be the default behavior of the linker, but maybe there's a reason the XML isn't used anywhere anymore.

**Comment by vitek-karas** [comment]
The aggressive trimming mode should be enabled in Blazor - where the size matters the most. In other verticals we so far didn't enable it because it's more likely to break things. Linker has some functionality to detect if the app will break with this mode - if it sees a type reference to the attribute in question it will produce a warning. But lot of times, the attribute consumption code doesn't have a type reference instead it enumerates all attributes and looks for it by name, which linker can't figure out. If this happens, the app may break.

Basically for now we decided not try to figure out a solid approach for trimming attributes. We maybe want to look into that.

I agree that the trimmer attributes are very likely safe to be removed, but it's a general approach which for now we decided against.

I'm not against doing this for the trimmer attributes (and other Requires... attributes as well), but I think we should try to gather some data - how much size is it going to safe?
